### PR TITLE
add k8s 1.11 e2e

### LINKED ISSUE #2745: Add unit tests and e2e tests for k8s 1.11
Follow up on #2739

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #2745 

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [x] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/acsengine/k8s_versions.go [hunk 1]
```diff
@@ -8,6 +8,33 @@ import (
 )
 
 var k8sComponentVersions = map[string]map[string]string{
+	"1.11": {
```
Comment (CecileRobertMichon) [inline pkg/acsengine/k8s_versions.go]: @andyzhangx @feiskyer do you have any comments on k8s components compatibility with k8s 1.11?

**Comment by CecileRobertMichon** [comment]
Need to fix #2790. Currently kubelet fails to start on agent nodes with error ` 8346 server.go:164] unrecognized key: Accelerators`

File: pkg/acsengine/defaults-kubelet.go [hunk 2]
```diff
@@ -106,7 +106,9 @@ func setKubeletConfig(cs *api.ContainerService) {
 			profile.KubernetesConfig.KubeletConfig = copyMap(profile.KubernetesConfig.KubeletConfig)
 		}
 		setMissingKubeletValues(profile.KubernetesConfig, o.KubernetesConfig.KubeletConfig)
-		addDefaultFeatureGates(profile.KubernetesConfig.KubeletConfig, o.OrchestratorVersion, "1.6.0", "Accelerators=true")
+		if !common.IsKubernetesVersionGe(o.OrchestratorVersion, "1.11.0") {
```
Comment (CecileRobertMichon) [inline pkg/acsengine/defaults-kubelet.go]: Will need to make sure this doesn't conflict with #2545

Comment (sozercan) [reply pkg/acsengine/defaults-kubelet.go]: thanks @CecileRobertMichon, shouldn't be a problem

**Comment by CecileRobertMichon** [comment]
Opened #2832 to address Windows support

**Pull request merged**